include/aws/cal/rsa.h

```diff
@@ -0,0 +1,167 @@
+#ifndef AWS_CAL_RSA_H
+#define AWS_CAL_RSA_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/cal/cal.h>
+#include <aws/common/byte_buf.h>
+
+AWS_PUSH_SANE_WARNING_LEVEL
+
+struct aws_rsa_key_pair;
+
+enum aws_rsa_encryption_algorithm {
+    AWS_CAL_RSA_ENCRYPTION_PKCS1_5,
+    AWS_CAL_RSA_ENCRYPTION_OAEP_SHA256,
+    AWS_CAL_RSA_ENCRYPTION_OAEP_SHA512,
+};
+
+enum aws_rsa_signing_algorithm { 
+    AWS_CAL_RSA_SIGNATURE_PKCS1_5_SHA256, 
```

Comment:
trivial: seems nicer if type name aligns with value names
```suggestion
enum aws_rsa_signature_algorithm { 
    AWS_CAL_RSA_SIGNATURE_PKCS1_5_SHA256, 
```